js-exercises/aperture/aperture.js

```diff
@@ -0,0 +1,26 @@
+function aperture(...args) {
+  let tupleCount = args[0];
+  let inputArray = args[1];
+  let arrayOfTuples = [];
+  let noOfTuples = inputArray.length - (tupleCount - 1);
+  let arrayOfItems = [];
+  let item = 0;
+
+  if(tupleCount > inputArray.length || inputArray.length === 0) {
+    return arrayOfTuples;
+  }
+
+  for(let i = 0; i < noOfTuples; i++) {
+    arrayOfItems = [];
+    item = i;
+    for(let j = 0; j < tupleCount; j++) {
+      arrayOfItems.push(inputArray[item]);
+      item++;
+    }
+    arrayOfTuples.push(arrayOfItems);
+  }
+
+  return arrayOfTuples;
```

Comment:
You could just loop through the given array, slice the required number of elements from it and push it to a new array.
Finally, return the new array.